tests/unit/test_configuration.py

```diff
@@ -1,7 +1,56 @@
+# TODO: type ignores and crawlee_storage_dir
+# https://github.com/apify/crawlee-python/issues/146
+
 from __future__ import annotations
 
+from typing import TYPE_CHECKING
+
 from crawlee.configuration import Configuration
+from crawlee.http_crawler import HttpCrawler, HttpCrawlingContext
+
+if TYPE_CHECKING:
+    from pathlib import Path
 
 
 def test_global_configuration_works() -> None:
     assert Configuration.get_global_configuration() is Configuration.get_global_configuration()
+
+
+async def test_storage_not_persisted_when_disabled(tmp_path: Path) -> None:
+    # Configure the crawler to not persist storage or metadata.
+    configuration = Configuration(
+        persist_storage=False,
+        write_metadata=False,
+        crawlee_storage_dir=str(tmp_path),  # type: ignore
+    )
+
+    crawler = HttpCrawler(configuration=configuration)
```

Comment:
Isn't it strange that `persist_storage` can be set normally here?

Reply:
for further context: https://github.com/apify/crawlee-python/issues/146